scipy/special/special/hyp2f1.h

```diff
@@ -0,0 +1,675 @@
+        }
+    }
+    // Kummer's Theorem for z = -1; c = 1 + a - b (DLMF 15.4.26)
+    if (std::abs(z + 1.0) < detail::hyp2f1_EPS && std::abs(1 + a - b - c) < detail::hyp2f1_EPS && !c_non_pos_int) {
+        return detail::four_gammas(a - b + 1, 0.5 * a + 1, a + 1, 0.5 * a - b + 1);
+    }
+    std::complex<double> result;
+    bool c_minus_a_neg_int = c - a == std::trunc(c - a) && c - a < 0;
+    bool c_minus_b_neg_int = c - b == std::trunc(c - b) && c - b < 0;
+    /* If one of c - a or c - b is a negative integer, reduces to evaluating
+     * a polynomial through an Euler hypergeometric transformation.
+     * (DLMF 15.8.1) */
+    if (c_minus_a_neg_int || c_minus_b_neg_int) {
+        max_degree = c_minus_b_neg_int ? std::abs(c - b) : std::abs(c - a);
+        if (max_degree <= UINT64_MAX) {
+            result = std::pow(1.0 - z, c - a - b);
+            auto series_generator = detail::HypergeometricSeriesGenerator(c - a, c - b, c, z);
+            result *=
+                detail::series_eval_fixed_length(series_generator, std::complex<double>{0.0, 0.0}, max_degree + 2);
+            return result;
+        } else {
+            set_error("hyp2f1", SF_ERROR_NO_RESULT, NULL);
+            return std::complex<double>{std::numeric_limits<double>::quiet_NaN(),
+                                        std::numeric_limits<double>::quiet_NaN()};
+        }
+    }
+    /* Diverges as real(z) -> 1 when c <= a + b.
+     * Todo: Actually check for overflow instead of using a fixed tolerance for
+     * all parameter combinations like in the Fortran original. */
+    if (std::abs(1 - z.real()) < detail::hyp2f1_EPS && z.imag() == 0 && c - a - b <= 0 && !c_non_pos_int) {
+        return std::complex<double>{std::numeric_limits<double>::infinity(), 0};
+    }
+    // Gauss's Summation Theorem for z = 1; c - a - b > 0 (DLMF 15.4.20).
+    if (z == 1.0 && c - a - b > 0 && !c_non_pos_int) {
+        return detail::four_gammas(c, c - a - b, c - a, c - b);
+    }
+    /* |z| < 0, z.real() >= 0. Use the Maclaurin Series.
+     * -----------------------------------------------------------------------
+     * Apply Euler Hypergeometric Transformation (DLMF 15.8.1) to reduce
+     * size of a and b if possible. We follow Zhang and Jin's
+     * implementation [1] although there is very likely a better heuristic
+     * to determine when this transformation should be applied. As it
+     * stands, this hurts precision in some cases. */
+    if (z_abs < 0.9 && z.real() >= 0) {
+        if (c - a < a && c - b < b) {
+            result = std::pow(1.0 - z, c - a - b);
+            // Maximum number of terms 1500 comes from Zhang and Jin.
+            auto series_generator = detail::HypergeometricSeriesGenerator(c - a, c - b, c, z);
+            result *= detail::series_eval(series_generator, std::complex<double>{0.0, 0.0}, detail::hyp2f1_EPS,
+                                          detail::hyp2f1_MAXITER, "hyp2f1");
```

Comment:
@steppi IIUC the comment says `1500` is the max terms here but `detail::hyp2f1_MAXITER` is `3000`, did I miss something, or is one of the comment of the `max_iter` argument value misstated?

Reply:
spoke on slack the comment was simply out of date from findings on refactor. Some cases weren't converging at 1500 even though the book indicated 1500 max_iter.